- en: What’s new in 1.1.4 (October 30, 2020)
  id: totrans-0
  prefs:
  - PREF_H1
  type: TYPE_NORMAL
  zh: 1.1.4 新内容（2020 年 10 月 30 日）
- en: 原文：[https://pandas.pydata.org/docs/whatsnew/v1.1.4.html](https://pandas.pydata.org/docs/whatsnew/v1.1.4.html)
  id: totrans-1
  prefs:
  - PREF_BQ
  type: TYPE_NORMAL
  zh: 原文：[https://pandas.pydata.org/docs/whatsnew/v1.1.4.html](https://pandas.pydata.org/docs/whatsnew/v1.1.4.html)
- en: These are the changes in pandas 1.1.4\. See [Release notes](index.html#release)
    for a full changelog including other versions of pandas.
  id: totrans-2
  prefs: []
  type: TYPE_NORMAL
  zh: 这些是 pandas 1.1.4 版本的更改。查看[发布说明](index.html#release)以获取包括其他 pandas 版本的完整更改日志。
- en: '## Fixed regressions'
  id: totrans-3
  prefs: []
  type: TYPE_NORMAL
  zh: '## 修复的回归问题'
- en: Fixed regression in [`read_csv()`](../reference/api/pandas.read_csv.html#pandas.read_csv
    "pandas.read_csv") raising a `ValueError` when `names` was of type `dict_keys`
    ([GH 36928](https://github.com/pandas-dev/pandas/issues/36928))
  id: totrans-4
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了当`names`为`dict_keys`类型时，[`read_csv()`](../reference/api/pandas.read_csv.html#pandas.read_csv
    "pandas.read_csv")抛出`ValueError`的回归问题 ([GH 36928](https://github.com/pandas-dev/pandas/issues/36928))
- en: Fixed regression in [`read_csv()`](../reference/api/pandas.read_csv.html#pandas.read_csv
    "pandas.read_csv") with more than 1M rows and specifying a `index_col` argument
    ([GH 37094](https://github.com/pandas-dev/pandas/issues/37094))
  id: totrans-5
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`read_csv()`](../reference/api/pandas.read_csv.html#pandas.read_csv "pandas.read_csv")在指定了`index_col`参数且超过
    100 万行时的回归问题 ([GH 37094](https://github.com/pandas-dev/pandas/issues/37094))
- en: Fixed regression where attempting to mutate a `DateOffset` object would no longer
    raise an `AttributeError` ([GH 36940](https://github.com/pandas-dev/pandas/issues/36940))
  id: totrans-6
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了尝试变异`DateOffset`对象时不再引发`AttributeError`的回归问题 ([GH 36940](https://github.com/pandas-dev/pandas/issues/36940))
- en: Fixed regression where [`DataFrame.agg()`](../reference/api/pandas.DataFrame.agg.html#pandas.DataFrame.agg
    "pandas.DataFrame.agg") would fail with [`TypeError`](https://docs.python.org/3/library/exceptions.html#TypeError
    "(in Python v3.12)") when passed positional arguments to be passed on to the aggregation
    function ([GH 36948](https://github.com/pandas-dev/pandas/issues/36948)).
  id: totrans-7
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了当传递位置参数以传递给聚合函数时，[`DataFrame.agg()`](../reference/api/pandas.DataFrame.agg.html#pandas.DataFrame.agg
    "pandas.DataFrame.agg")会失败的回归问题 ([GH 36948](https://github.com/pandas-dev/pandas/issues/36948))
- en: Fixed regression in `RollingGroupby` with `sort=False` not being respected ([GH
    36889](https://github.com/pandas-dev/pandas/issues/36889))
  id: totrans-8
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了带有`sort=False`参数的`RollingGroupby`中未能被尊重的回归问题 ([GH 36889](https://github.com/pandas-dev/pandas/issues/36889))
- en: Fixed regression in [`Series.astype()`](../reference/api/pandas.Series.astype.html#pandas.Series.astype
    "pandas.Series.astype") converting `None` to `"nan"` when casting to string ([GH
    36904](https://github.com/pandas-dev/pandas/issues/36904))
  id: totrans-9
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`Series.astype()`](../reference/api/pandas.Series.astype.html#pandas.Series.astype
    "pandas.Series.astype")方法在将`None`转换为`"nan"`时的回归问题 ([GH 36904](https://github.com/pandas-dev/pandas/issues/36904))
- en: Fixed regression in [`Series.rank()`](../reference/api/pandas.Series.rank.html#pandas.Series.rank
    "pandas.Series.rank") method failing for read-only data ([GH 37290](https://github.com/pandas-dev/pandas/issues/37290))
  id: totrans-10
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了对只读数据的[`Series.rank()`](../reference/api/pandas.Series.rank.html#pandas.Series.rank
    "pandas.Series.rank")方法失败的回归问题 ([GH 37290](https://github.com/pandas-dev/pandas/issues/37290))
- en: Fixed regression in `RollingGroupby` causing a segmentation fault with Index
    of dtype object ([GH 36727](https://github.com/pandas-dev/pandas/issues/36727))
  id: totrans-11
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了在`RollingGroupby`中导致索引为对象dtype时段错误的回归问题 ([GH 36727](https://github.com/pandas-dev/pandas/issues/36727))
- en: Fixed regression in `DataFrame.resample(...).apply(...)()` raised `AttributeError`
    when input was a [`DataFrame`](../reference/api/pandas.DataFrame.html#pandas.DataFrame
    "pandas.DataFrame") and only a [`Series`](../reference/api/pandas.Series.html#pandas.Series
    "pandas.Series") was evaluated ([GH 36951](https://github.com/pandas-dev/pandas/issues/36951))
  id: totrans-12
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了当输入为[`DataFrame`](../reference/api/pandas.DataFrame.html#pandas.DataFrame
    "pandas.DataFrame")且仅评估了[`Series`](../reference/api/pandas.Series.html#pandas.Series
    "pandas.Series")时，`DataFrame.resample(...).apply(...)()`引发`AttributeError`的回归问题
    ([GH 36951](https://github.com/pandas-dev/pandas/issues/36951))
- en: Fixed regression in `DataFrame.groupby(..).std()` with nullable integer dtype
    ([GH 37415](https://github.com/pandas-dev/pandas/issues/37415))
  id: totrans-13
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了带有可空整数dtype的`DataFrame.groupby(..).std()`中的回归问题 ([GH 37415](https://github.com/pandas-dev/pandas/issues/37415))
- en: Fixed regression in [`PeriodDtype`](../reference/api/pandas.PeriodDtype.html#pandas.PeriodDtype
    "pandas.PeriodDtype") comparing both equal and unequal to its string representation
    ([GH 37265](https://github.com/pandas-dev/pandas/issues/37265))
  id: totrans-14
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`PeriodDtype`](../reference/api/pandas.PeriodDtype.html#pandas.PeriodDtype
    "pandas.PeriodDtype")与其字符串表示形式的相等性和不相等性比较的回归问题 ([GH 37265](https://github.com/pandas-dev/pandas/issues/37265))
- en: Fixed regression where slicing [`DatetimeIndex`](../reference/api/pandas.DatetimeIndex.html#pandas.DatetimeIndex
    "pandas.DatetimeIndex") raised [`AssertionError`](https://docs.python.org/3/library/exceptions.html#AssertionError
    "(in Python v3.12)") on irregular time series with `pd.NaT` or on unsorted indices
    ([GH 36953](https://github.com/pandas-dev/pandas/issues/36953) and [GH 35509](https://github.com/pandas-dev/pandas/issues/35509))
  id: totrans-15
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了切片[`DatetimeIndex`](../reference/api/pandas.DatetimeIndex.html#pandas.DatetimeIndex
    "pandas.DatetimeIndex")在不规则时间序列上出现 [`AssertionError`](https://docs.python.org/3/library/exceptions.html#AssertionError
    "(in Python v3.12)") 或在未排序索引上时引发的回归问题，带有 `pd.NaT` ([GH 36953](https://github.com/pandas-dev/pandas/issues/36953))
    或未排序索引 ([GH 35509](https://github.com/pandas-dev/pandas/issues/35509))
- en: Fixed regression in certain offsets ([`pd.offsets.Day()`](../reference/api/pandas.tseries.offsets.Day.html#pandas.tseries.offsets.Day
    "pandas.tseries.offsets.Day") and below) no longer being hashable ([GH 37267](https://github.com/pandas-dev/pandas/issues/37267))
  id: totrans-16
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了某些偏移量的回归问题（[`pd.offsets.Day()`](../reference/api/pandas.tseries.offsets.Day.html#pandas.tseries.offsets.Day
    "pandas.tseries.offsets.Day")及以下），不再可散列 ([GH 37267](https://github.com/pandas-dev/pandas/issues/37267))
- en: Fixed regression in `StataReader` which required `chunksize` to be manually
    set when using an iterator to read a dataset ([GH 37280](https://github.com/pandas-dev/pandas/issues/37280))
  id: totrans-17
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了 `StataReader` 中的回归，当使用迭代器读取数据集时，需要手动设置 `chunksize` ([GH 37280](https://github.com/pandas-dev/pandas/issues/37280))
- en: Fixed regression in setitem with [`DataFrame.iloc()`](../reference/api/pandas.DataFrame.iloc.html#pandas.DataFrame.iloc
    "pandas.DataFrame.iloc") which raised error when trying to set a value while filtering
    with a boolean list ([GH 36741](https://github.com/pandas-dev/pandas/issues/36741))
  id: totrans-18
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了与 [`DataFrame.iloc()`](../reference/api/pandas.DataFrame.iloc.html#pandas.DataFrame.iloc
    "pandas.DataFrame.iloc") 结合使用时的 setitem 回归问题，尝试在使用布尔列表进行筛选时设置值时引发错误 ([GH 36741](https://github.com/pandas-dev/pandas/issues/36741))
- en: Fixed regression in setitem with a Series getting aligned before setting the
    values ([GH 37427](https://github.com/pandas-dev/pandas/issues/37427))
  id: totrans-19
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了在将 Series 与 setitem 结合使用时，在设置值之前对齐的回归问题 ([GH 37427](https://github.com/pandas-dev/pandas/issues/37427))
- en: Fixed regression in `MultiIndex.is_monotonic_increasing` returning wrong results
    with `NaN` in at least one of the levels ([GH 37220](https://github.com/pandas-dev/pandas/issues/37220))
  id: totrans-20
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了 `MultiIndex.is_monotonic_increasing` 在至少一个级别中出现 `NaN` 时返回错误结果的回归问题 ([GH
    37220](https://github.com/pandas-dev/pandas/issues/37220))
- en: 'Fixed regression in inplace arithmetic operation (+=) on a Series not updating
    the parent DataFrame/Series ([GH 36373](https://github.com/pandas-dev/pandas/issues/36373))  ##
    Bug fixes'
  id: totrans-21
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: '修复了对 Series 的就地算术操作（+=）不更新父 DataFrame/Series 的回归问题 ([GH 36373](https://github.com/pandas-dev/pandas/issues/36373))  ##
    问题修复'
- en: Bug causing `groupby(...).sum()` and similar to not preserve metadata ([GH 29442](https://github.com/pandas-dev/pandas/issues/29442))
  id: totrans-22
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Bug causing `groupby(...).sum()` 和类似操作不保留元数据 ([GH 29442](https://github.com/pandas-dev/pandas/issues/29442))
- en: Bug in [`Series.isin()`](../reference/api/pandas.Series.isin.html#pandas.Series.isin
    "pandas.Series.isin") and [`DataFrame.isin()`](../reference/api/pandas.DataFrame.isin.html#pandas.DataFrame.isin
    "pandas.DataFrame.isin") raising a `ValueError` when the target was read-only
    ([GH 37174](https://github.com/pandas-dev/pandas/issues/37174))
  id: totrans-23
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: '[`Series.isin()`](../reference/api/pandas.Series.isin.html#pandas.Series.isin
    "pandas.Series.isin") 和 [`DataFrame.isin()`](../reference/api/pandas.DataFrame.isin.html#pandas.DataFrame.isin
    "pandas.DataFrame.isin") 中的 Bug 在目标为只读时引发 `ValueError` ([GH 37174](https://github.com/pandas-dev/pandas/issues/37174))'
- en: Bug in [`DataFrameGroupBy.fillna()`](../reference/api/pandas.core.groupby.DataFrameGroupBy.fillna.html#pandas.core.groupby.DataFrameGroupBy.fillna
    "pandas.core.groupby.DataFrameGroupBy.fillna") and [`SeriesGroupBy.fillna()`](../reference/api/pandas.core.groupby.SeriesGroupBy.fillna.html#pandas.core.groupby.SeriesGroupBy.fillna
    "pandas.core.groupby.SeriesGroupBy.fillna") that introduced a performance regression
    after 1.0.5 ([GH 36757](https://github.com/pandas-dev/pandas/issues/36757))
  id: totrans-24
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: '[`DataFrameGroupBy.fillna()`](../reference/api/pandas.core.groupby.DataFrameGroupBy.fillna.html#pandas.core.groupby.DataFrameGroupBy.fillna
    "pandas.core.groupby.DataFrameGroupBy.fillna") 和 [`SeriesGroupBy.fillna()`](../reference/api/pandas.core.groupby.SeriesGroupBy.fillna.html#pandas.core.groupby.SeriesGroupBy.fillna
    "pandas.core.groupby.SeriesGroupBy.fillna") 中的 Bug 在 1.0.5 之后引入了性能回归 ([GH 36757](https://github.com/pandas-dev/pandas/issues/36757))'
- en: Bug in [`DataFrame.info()`](../reference/api/pandas.DataFrame.info.html#pandas.DataFrame.info
    "pandas.DataFrame.info") was raising a `KeyError` when the DataFrame has integer
    column names ([GH 37245](https://github.com/pandas-dev/pandas/issues/37245))
  id: totrans-25
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: '[`DataFrame.info()`](../reference/api/pandas.DataFrame.info.html#pandas.DataFrame.info
    "pandas.DataFrame.info") 中的 Bug 在 DataFrame 具有整数列名时引发 `KeyError` ([GH 37245](https://github.com/pandas-dev/pandas/issues/37245))'
- en: 'Bug in `DataFrameGroupby.apply()` would drop a [`CategoricalIndex`](../reference/api/pandas.CategoricalIndex.html#pandas.CategoricalIndex
    "pandas.CategoricalIndex") when grouped on ([GH 35792](https://github.com/pandas-dev/pandas/issues/35792))  ##
    Contributors'
  id: totrans-26
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: '`DataFrameGroupby.apply()`中的错误会在分组时删除[`CategoricalIndex`](../reference/api/pandas.CategoricalIndex.html#pandas.CategoricalIndex
    "pandas.CategoricalIndex")（[GH 35792](https://github.com/pandas-dev/pandas/issues/35792）  ##
    贡献者'
- en: A total of 18 people contributed patches to this release. People with a “+”
    by their names contributed a patch for the first time.
  id: totrans-27
  prefs: []
  type: TYPE_NORMAL
  zh: 总共有18人为此版本贡献了补丁。名字后面带有“+”的人第一次贡献了补丁。
- en: Daniel Saxton
  id: totrans-28
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Daniel Saxton
- en: Fangchen Li
  id: totrans-29
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Fangchen Li
- en: Janus +
  id: totrans-30
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Janus +
- en: Joris Van den Bossche
  id: totrans-31
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Joris Van den Bossche
- en: Kevin Sheppard
  id: totrans-32
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Kevin Sheppard
- en: Marco Gorelli
  id: totrans-33
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Marco Gorelli
- en: Matt Roeschke
  id: totrans-34
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Matt Roeschke
- en: Matthew Roeschke
  id: totrans-35
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Matthew Roeschke
- en: MeeseeksMachine
  id: totrans-36
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: MeeseeksMachine
- en: Pandas Development Team
  id: totrans-37
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Pandas 开发团队
- en: Paul Ganssle
  id: totrans-38
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Paul Ganssle
- en: Richard Shadrach
  id: totrans-39
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Richard Shadrach
- en: Simon Hawkins
  id: totrans-40
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Simon Hawkins
- en: Thomas Smith
  id: totrans-41
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Thomas Smith
- en: Tobias Pitters
  id: totrans-42
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Tobias Pitters
- en: abmyii +
  id: totrans-43
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: abmyii +
- en: jbrockmendel
  id: totrans-44
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: jbrockmendel
- en: 'patrick  ## Fixed regressions'
  id: totrans-45
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 'patrick  ## 修复的回归问题'
- en: Fixed regression in [`read_csv()`](../reference/api/pandas.read_csv.html#pandas.read_csv
    "pandas.read_csv") raising a `ValueError` when `names` was of type `dict_keys`
    ([GH 36928](https://github.com/pandas-dev/pandas/issues/36928))
  id: totrans-46
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`read_csv()`](../reference/api/pandas.read_csv.html#pandas.read_csv "pandas.read_csv")方法在`names`为`dict_keys`类型时引发`ValueError`的回归问题（[GH
    36928](https://github.com/pandas-dev/pandas/issues/36928)）
- en: Fixed regression in [`read_csv()`](../reference/api/pandas.read_csv.html#pandas.read_csv
    "pandas.read_csv") with more than 1M rows and specifying a `index_col` argument
    ([GH 37094](https://github.com/pandas-dev/pandas/issues/37094))
  id: totrans-47
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`read_csv()`](../reference/api/pandas.read_csv.html#pandas.read_csv "pandas.read_csv")在超过100万行并指定`index_col`参数时的回归问题（[GH
    37094](https://github.com/pandas-dev/pandas/issues/37094)）
- en: Fixed regression where attempting to mutate a `DateOffset` object would no longer
    raise an `AttributeError` ([GH 36940](https://github.com/pandas-dev/pandas/issues/36940))
  id: totrans-48
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了尝试改变`DateOffset`对象时不再引发`AttributeError`的回归问题（[GH 36940](https://github.com/pandas-dev/pandas/issues/36940)）
- en: Fixed regression where [`DataFrame.agg()`](../reference/api/pandas.DataFrame.agg.html#pandas.DataFrame.agg
    "pandas.DataFrame.agg") would fail with [`TypeError`](https://docs.python.org/3/library/exceptions.html#TypeError
    "(in Python v3.12)") when passed positional arguments to be passed on to the aggregation
    function ([GH 36948](https://github.com/pandas-dev/pandas/issues/36948)).
  id: totrans-49
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`DataFrame.agg()`](../reference/api/pandas.DataFrame.agg.html#pandas.DataFrame.agg
    "pandas.DataFrame.agg")在传递给聚合函数的位置参数时会失败并引发[`TypeError`](https://docs.python.org/3/library/exceptions.html#TypeError
    "(在 Python v3.12 中)")的回归问题（[GH 36948](https://github.com/pandas-dev/pandas/issues/36948))。
- en: Fixed regression in `RollingGroupby` with `sort=False` not being respected ([GH
    36889](https://github.com/pandas-dev/pandas/issues/36889))
  id: totrans-50
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了`RollingGroupby`中`sort=False`未被尊重的回归问题（[GH 36889](https://github.com/pandas-dev/pandas/issues/36889)）
- en: Fixed regression in [`Series.astype()`](../reference/api/pandas.Series.astype.html#pandas.Series.astype
    "pandas.Series.astype") converting `None` to `"nan"` when casting to string ([GH
    36904](https://github.com/pandas-dev/pandas/issues/36904))
  id: totrans-51
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`Series.astype()`](../reference/api/pandas.Series.astype.html#pandas.Series.astype
    "pandas.Series.astype")将`None`转换为`"nan"`时转换为字符串时的回���问题（[GH 36904](https://github.com/pandas-dev/pandas/issues/36904)）
- en: Fixed regression in [`Series.rank()`](../reference/api/pandas.Series.rank.html#pandas.Series.rank
    "pandas.Series.rank") method failing for read-only data ([GH 37290](https://github.com/pandas-dev/pandas/issues/37290))
  id: totrans-52
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`Series.rank()`](../reference/api/pandas.Series.rank.html#pandas.Series.rank
    "pandas.Series.rank")方法在只读数据上失败的回归问题（[GH 37290](https://github.com/pandas-dev/pandas/issues/37290)）
- en: Fixed regression in `RollingGroupby` causing a segmentation fault with Index
    of dtype object ([GH 36727](https://github.com/pandas-dev/pandas/issues/36727))
  id: totrans-53
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了`RollingGroupby`在对象dtype的索引下导致分段错误的回归问题（[GH 36727](https://github.com/pandas-dev/pandas/issues/36727)）
- en: Fixed regression in `DataFrame.resample(...).apply(...)()` raised `AttributeError`
    when input was a [`DataFrame`](../reference/api/pandas.DataFrame.html#pandas.DataFrame
    "pandas.DataFrame") and only a [`Series`](../reference/api/pandas.Series.html#pandas.Series
    "pandas.Series") was evaluated ([GH 36951](https://github.com/pandas-dev/pandas/issues/36951))
  id: totrans-54
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了`DataFrame.resample(...).apply(...)()`在输入为[`DataFrame`](../reference/api/pandas.DataFrame.html#pandas.DataFrame
    "pandas.DataFrame")且仅评估[`Series`](../reference/api/pandas.Series.html#pandas.Series
    "pandas.Series")时引发`AttributeError`的回归问题（[GH 36951](https://github.com/pandas-dev/pandas/issues/36951)）
- en: Fixed regression in `DataFrame.groupby(..).std()` with nullable integer dtype
    ([GH 37415](https://github.com/pandas-dev/pandas/issues/37415))
  id: totrans-55
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了带有可空整数dtype的`DataFrame.groupby(..).std()`的回归问题（[GH 37415](https://github.com/pandas-dev/pandas/issues/37415)）
- en: Fixed regression in [`PeriodDtype`](../reference/api/pandas.PeriodDtype.html#pandas.PeriodDtype
    "pandas.PeriodDtype") comparing both equal and unequal to its string representation
    ([GH 37265](https://github.com/pandas-dev/pandas/issues/37265))
  id: totrans-56
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了[`PeriodDtype`](../reference/api/pandas.PeriodDtype.html#pandas.PeriodDtype
    "pandas.PeriodDtype")在比较相等和不相等时与其字符串表示形式的回归问题（[GH 37265](https://github.com/pandas-dev/pandas/issues/37265))
- en: Fixed regression where slicing [`DatetimeIndex`](../reference/api/pandas.DatetimeIndex.html#pandas.DatetimeIndex
    "pandas.DatetimeIndex") raised [`AssertionError`](https://docs.python.org/3/library/exceptions.html#AssertionError
    "(in Python v3.12)") on irregular time series with `pd.NaT` or on unsorted indices
    ([GH 36953](https://github.com/pandas-dev/pandas/issues/36953) and [GH 35509](https://github.com/pandas-dev/pandas/issues/35509))
  id: totrans-57
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了在不规则时间序列上切片 [`DatetimeIndex`](../reference/api/pandas.DatetimeIndex.html#pandas.DatetimeIndex
    "pandas.DatetimeIndex") 或未排序索引时引发 [`AssertionError`](https://docs.python.org/3/library/exceptions.html#AssertionError
    "(in Python v3.12)") 的回归问题（[GH 36953](https://github.com/pandas-dev/pandas/issues/36953)
    和 [GH 35509](https://github.com/pandas-dev/pandas/issues/35509))
- en: Fixed regression in certain offsets ([`pd.offsets.Day()`](../reference/api/pandas.tseries.offsets.Day.html#pandas.tseries.offsets.Day
    "pandas.tseries.offsets.Day") and below) no longer being hashable ([GH 37267](https://github.com/pandas-dev/pandas/issues/37267))
  id: totrans-58
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了某些偏移量（[`pd.offsets.Day()`](../reference/api/pandas.tseries.offsets.Day.html#pandas.tseries.offsets.Day
    "pandas.tseries.offsets.Day")及以下）不再可哈希的回归问题（[GH 37267](https://github.com/pandas-dev/pandas/issues/37267))
- en: Fixed regression in `StataReader` which required `chunksize` to be manually
    set when using an iterator to read a dataset ([GH 37280](https://github.com/pandas-dev/pandas/issues/37280))
  id: totrans-59
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了使用迭代器读取数据集时，需要手动设置 `chunksize` 的 `StataReader` 的回归问题（[GH 37280](https://github.com/pandas-dev/pandas/issues/37280))
- en: Fixed regression in setitem with [`DataFrame.iloc()`](../reference/api/pandas.DataFrame.iloc.html#pandas.DataFrame.iloc
    "pandas.DataFrame.iloc") which raised error when trying to set a value while filtering
    with a boolean list ([GH 36741](https://github.com/pandas-dev/pandas/issues/36741))
  id: totrans-60
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了在使用布尔列表过滤时尝试设置值时引发错误的 [`DataFrame.iloc()`](../reference/api/pandas.DataFrame.iloc.html#pandas.DataFrame.iloc
    "pandas.DataFrame.iloc") 的回归问题（[GH 36741](https://github.com/pandas-dev/pandas/issues/36741))
- en: Fixed regression in setitem with a Series getting aligned before setting the
    values ([GH 37427](https://github.com/pandas-dev/pandas/issues/37427))
  id: totrans-61
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了设置一系列值之前对齐 Series 的回归问题（[GH 37427](https://github.com/pandas-dev/pandas/issues/37427))
- en: Fixed regression in `MultiIndex.is_monotonic_increasing` returning wrong results
    with `NaN` in at least one of the levels ([GH 37220](https://github.com/pandas-dev/pandas/issues/37220))
  id: totrans-62
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了 `MultiIndex.is_monotonic_increasing` 在至少一个级别中存在 `NaN` 时返回错误结果的回归问题（[GH 37220](https://github.com/pandas-dev/pandas/issues/37220))
- en: Fixed regression in inplace arithmetic operation (+=) on a Series not updating
    the parent DataFrame/Series ([GH 36373](https://github.com/pandas-dev/pandas/issues/36373))
  id: totrans-63
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 修复了对 Series 进行就地算术运算（+=）时未更新父 DataFrame/Series 的回归问题（[GH 36373](https://github.com/pandas-dev/pandas/issues/36373))
- en: '## Bug fixes'
  id: totrans-64
  prefs: []
  type: TYPE_NORMAL
  zh: '## Bug fixes'
- en: Bug causing `groupby(...).sum()` and similar to not preserve metadata ([GH 29442](https://github.com/pandas-dev/pandas/issues/29442))
  id: totrans-65
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 导致 `groupby(...).sum()` 和类似方法不保留元数据的错误已修复（[GH 29442](https://github.com/pandas-dev/pandas/issues/29442))
- en: Bug in [`Series.isin()`](../reference/api/pandas.Series.isin.html#pandas.Series.isin
    "pandas.Series.isin") and [`DataFrame.isin()`](../reference/api/pandas.DataFrame.isin.html#pandas.DataFrame.isin
    "pandas.DataFrame.isin") raising a `ValueError` when the target was read-only
    ([GH 37174](https://github.com/pandas-dev/pandas/issues/37174))
  id: totrans-66
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 当目标是只读时，[`Series.isin()`](../reference/api/pandas.Series.isin.html#pandas.Series.isin
    "pandas.Series.isin") 和 [`DataFrame.isin()`](../reference/api/pandas.DataFrame.isin.html#pandas.DataFrame.isin
    "pandas.DataFrame.isin") 引发 `ValueError` 的错误已修复（[GH 37174](https://github.com/pandas-dev/pandas/issues/37174))
- en: Bug in [`DataFrameGroupBy.fillna()`](../reference/api/pandas.core.groupby.DataFrameGroupBy.fillna.html#pandas.core.groupby.DataFrameGroupBy.fillna
    "pandas.core.groupby.DataFrameGroupBy.fillna") and [`SeriesGroupBy.fillna()`](../reference/api/pandas.core.groupby.SeriesGroupBy.fillna.html#pandas.core.groupby.SeriesGroupBy.fillna
    "pandas.core.groupby.SeriesGroupBy.fillna") that introduced a performance regression
    after 1.0.5 ([GH 36757](https://github.com/pandas-dev/pandas/issues/36757))
  id: totrans-67
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: '[`DataFrameGroupBy.fillna()`](../reference/api/pandas.core.groupby.DataFrameGroupBy.fillna.html#pandas.core.groupby.DataFrameGroupBy.fillna
    "pandas.core.groupby.DataFrameGroupBy.fillna") 和 [`SeriesGroupBy.fillna()`](../reference/api/pandas.core.groupby.SeriesGroupBy.fillna.html#pandas.core.groupby.SeriesGroupBy.fillna
    "pandas.core.groupby.SeriesGroupBy.fillna") 中引入的性能回归问题已在 1.0.5 之后修复（[GH 36757](https://github.com/pandas-dev/pandas/issues/36757))'
- en: Bug in [`DataFrame.info()`](../reference/api/pandas.DataFrame.info.html#pandas.DataFrame.info
    "pandas.DataFrame.info") was raising a `KeyError` when the DataFrame has integer
    column names ([GH 37245](https://github.com/pandas-dev/pandas/issues/37245))
  id: totrans-68
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 在 [`DataFrame.info()`](../reference/api/pandas.DataFrame.info.html#pandas.DataFrame.info
    "pandas.DataFrame.info") 中存在的 Bug，在 DataFrame 有整数列名时引发了 `KeyError` ([GH 37245](https://github.com/pandas-dev/pandas/issues/37245))
- en: Bug in `DataFrameGroupby.apply()` would drop a [`CategoricalIndex`](../reference/api/pandas.CategoricalIndex.html#pandas.CategoricalIndex
    "pandas.CategoricalIndex") when grouped on ([GH 35792](https://github.com/pandas-dev/pandas/issues/35792))
  id: totrans-69
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: 在 `DataFrameGroupby.apply()` 中存在的 Bug 会在分组时丢弃 [`CategoricalIndex`](../reference/api/pandas.CategoricalIndex.html#pandas.CategoricalIndex
    "pandas.CategoricalIndex") ([GH 35792](https://github.com/pandas-dev/pandas/issues/35792))
- en: '## Contributors'
  id: totrans-70
  prefs: []
  type: TYPE_NORMAL
  zh: '## 贡献者'
- en: A total of 18 people contributed patches to this release. People with a “+”
    by their names contributed a patch for the first time.
  id: totrans-71
  prefs: []
  type: TYPE_NORMAL
  zh: 总共有 18 人对此版本进行了贡献。名字后面带有“+”的人第一次贡献了一个补丁。
- en: Daniel Saxton
  id: totrans-72
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Daniel Saxton
- en: Fangchen Li
  id: totrans-73
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Fangchen Li
- en: Janus +
  id: totrans-74
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Janus +
- en: Joris Van den Bossche
  id: totrans-75
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Joris Van den Bossche
- en: Kevin Sheppard
  id: totrans-76
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Kevin Sheppard
- en: Marco Gorelli
  id: totrans-77
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Marco Gorelli
- en: Matt Roeschke
  id: totrans-78
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Matt Roeschke
- en: Matthew Roeschke
  id: totrans-79
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Matthew Roeschke
- en: MeeseeksMachine
  id: totrans-80
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: MeeseeksMachine
- en: Pandas Development Team
  id: totrans-81
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Pandas 开发团队
- en: Paul Ganssle
  id: totrans-82
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Paul Ganssle
- en: Richard Shadrach
  id: totrans-83
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Richard Shadrach
- en: Simon Hawkins
  id: totrans-84
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Simon Hawkins
- en: Thomas Smith
  id: totrans-85
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Thomas Smith
- en: Tobias Pitters
  id: totrans-86
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: Tobias Pitters
- en: abmyii +
  id: totrans-87
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: abmyii +
- en: jbrockmendel
  id: totrans-88
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: jbrockmendel
- en: patrick
  id: totrans-89
  prefs:
  - PREF_UL
  type: TYPE_NORMAL
  zh: patrick
